test/torch/tensors/test_additive_shared.py

```diff
@@ -187,3 +273,97 @@ def test_comp(workers):
+
+    t = torch.tensor([3, 1.0, 2])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    max_value = x.max().get().float_prec()
+    assert max_value == torch.tensor([3.0])
+
+    t = torch.tensor([3, 4.0])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    max_value = x.max().get().float_prec()
+    assert max_value == torch.tensor([4.0])
+
+    t = torch.tensor([3, 4.0, 5, 2])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    max_value = x.max().get().float_prec()
+    assert max_value == torch.tensor([5.0])
+
+
+def test_argmax(workers):
+    alice, bob, james = workers["alice"], workers["bob"], workers["james"]
+
+    t = torch.tensor([3, 1.0, 2])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    idx = x.argmax().get().float_prec()
+    assert idx == torch.tensor([0.0])
+
+    t = torch.tensor([3, 4.0])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    idx = x.argmax().get().float_prec()
+    assert idx == torch.tensor([1.0])
+
+    t = torch.tensor([3, 4.0, 5, 2])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    idx = x.argmax().get().float_prec()
+    assert idx == torch.tensor([2.0])
+
+    # no dim=
+    t = torch.tensor([[1, 2.0, 4], [3, 9.0, 2.0]])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    ids = x.argmax().get().float_prec()
+    assert ids.long() == torch.argmax(t)  # TODO rm .long()
+
+    # dim=1
+    t = torch.tensor([[1, 2.0, 4], [3, 1.0, 2.0]])
+    x = t.fix_prec().share(bob, alice, crypto_provider=james)
+    ids = x.argmax(dim=1).get().float_prec()
+    assert (ids.long() == torch.argmax(t, dim=1)).all()  # TODO rm .long()
+
+
+def test_mod(workers):
+    alice, bob, james = workers["alice"], workers["bob"], workers["james"]
+
+    t = torch.tensor([21]).share(bob, alice, crypto_provider=james)
+    assert t.child.mod(8).get() % 8 == torch.tensor([5])
+    assert t.child.mod(-8).get() % -8 == torch.tensor([-3])
+
+    t = torch.tensor([-21]).share(bob, alice, crypto_provider=james)
+    assert t.child.mod(8).get() % 8 == torch.tensor([3])
+    assert t.child.mod(-8).get() % -8 == torch.tensor([-5])
+
+    assert (t.child % 8).get() % 8 == torch.tensor([3])
+
+
+def test_unbind(workers):
+    alice, bob, james = workers["alice"], workers["bob"], workers["james"]
+
+    x = torch.tensor([21, 17]).share(bob, alice, crypto_provider=james).child
+
+    x0, x1 = torch.unbind(x)
+
+    assert x0.get() == torch.tensor(21)
+    assert x1.get() == torch.tensor(17)
+
+
+def test_handle_func_command(workers):
+    """
+    Just to show that handle_func_command works
+    Even is torch.abs should be hooked to return a correct value
+    """
+    alice, bob, james = workers["alice"], workers["bob"], workers["james"]
+
+    t = torch.tensor([-21]).share(bob, alice, crypto_provider=james).child
+    _ = torch.abs(t).get()
+
+
+def test_mul_with_no_crypto_provider(workers):
```

Comment:
this isnt mul should be test_init_with_no_crypto_provider

Reply:
Correct!